Bookish.Web/Controllers/HomeController.cs

```diff
@@ -65,12 +65,13 @@ public IActionResult NewBookPost(string title, string author, string isbn, int t
             }
 
             bookishService.AddBook(title, author, isbn, totalCopies);
-            return RedirectToAction("Barcodes");
+            return RedirectToAction("Barcodes", new { isbn, totalCopies });
         }
 
-        public IActionResult Barcodes()
+        public IActionResult Barcodes(string isbn, int totalCopies)
```

Comment:
I would expect isbn to be part of the url 🙂 And you can get totalCopies from the db, that also means you're verifying that the isbn is a valid book in the library 🙂

Reply:
To get totalCopies from the database which of the following should I do?
1. Reuse the existing GetCopies() method and use .Count() on the result, which means the query would return some redundant information.
2. Create a new method/query with a COUNT in the query.

Reply:
It depends. If you definitely just wanted the number of copies then I would create a new query. However you probably also want to display some metadata on your barcodes page, for example the title I think is worth displaying. And Shiv pointed out on his MR that the barcode should probably be different for each book, as it won't be used as a normal purchase barcode (the books should have that already) but instead as a way for the library to identify which book is being borrowed, returned. So you probably need the copy id as well. In that case I think you might as well use your existing method, as there's probably not a lot of redundant information at that point. What do you think? 🙂